### PR TITLE
feat(esign): Remove passphrase to avoid going into logging

### PR DESCRIPTION
**Pull request by fajarhikmal214**
## Overview
- feat(esign): Remove passphrase to avoid going into logging

## Evidence:
project: SIDEBAR
title: feat(esign): Remove passphrase to avoid going into logging
participants: @fajarhikmal214 @samudra-ajri @muhamadabduh @azophy 
screenshot: https://user-images.githubusercontent.com/79292118/224908138-ae42213a-b98c-4344-9311-e849c9827d23.png

### CODE REVIEW

**Review by samudra-ajri — APPROVED** [review]

**Review by muhamadabduh — APPROVED** [review]
why has it to be removed into logging ? is it because the passphrase is too sensitive to be exposed in the logging?

File: src/internal/esigns/usecase/usecase.ts [hunk 1]
```diff
@@ -158,7 +161,7 @@ class Usecase {
         const originalFileName = randomUUID() + '.pdf'
 
         const passphrase = passphraseDecryption(
-            body.passphrase,
+            body.passphrase!,
```
Comment (azophy) [inline src/internal/esigns/usecase/usecase.ts:164]: I'm not familiar with this syntax. Whats this line suppose to do? I would appreciate if you have learning resource about this 🙏

**Comment by azophy** [comment]
> why has it to be removed into logging ? is it because the passphrase is too sensitive to be exposed in the logging?

yes, 

> why has it to be removed into logging ? is it because the passphrase is too sensitive to be exposed in the logging?

yes kang. its dangerous as logs usually could be accessed by many people. allowing plain passphrase/password in log is equivalent as storing password as plaintext in database

File: src/internal/esigns/usecase/usecase.ts (hunk 1 above)
Comment (fajarhikmal214) [reply src/internal/esigns/usecase/usecase.ts:164]: It's called definite assignment assertion nodejs

According to CHAT GPT, Definite assignment assertion is a feature in TypeScript that allows you to tell the compiler that a variable is definitely assigned a value, even if the compiler cannot detect it.

Comment (azophy) [reply src/internal/esigns/usecase/usecase.ts:164]: nice2, thanks for the reference. found the official docs about it: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#definite-assignment-assertions

**Review by azophy — APPROVED** [review]

**Pull request merged**